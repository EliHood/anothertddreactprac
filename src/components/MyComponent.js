import React, {Component} from 'react';

export default class MyComponent extends Component{
    updateState(event){
        this.setState({
            input:event.target.value
        });
    }

    render(){
        return(
        <div>
            <input type="text" 
                    onChange={this.updateState.bind(this)} />
         </div>


        )
    }
}