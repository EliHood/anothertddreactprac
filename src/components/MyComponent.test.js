import React from 'react';
import { shallow } from 'enzyme';

import MyComponent from './MyComponent';

describe("My Component", ()=> {
    it("Should render the MyComponent component", ()=>{
        const wrapper = shallow(<MyComponent />)
    })


    it("Should render the initial layout", ()=>{
        const component = shallow(<MyComponent />);
        // then
        expect(component.getElements()).toMatchSnapshot();
    })

    it("should create an entry in component state", ()=>{
        const component = shallow(<MyComponent/>);
        // find class name 
        const form = component.find('input');

        form.props().onChange({target:{
            name: 'myName',
            value:'myValue'
        }});

        // simulate on change

        expect (component.state('input')).toEqual('myValue');
    })
})