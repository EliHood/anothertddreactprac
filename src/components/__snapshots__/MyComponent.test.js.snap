// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Component Should render the initial layout 1`] = `
Array [
  <div>
    <input
      onChange={[Function]}
      type="text"
    />
  </div>,
]
`;

exports[`MyComponent should render initial layout 1`] = `
Array [
<div>
    <input
         onChange={[Function]}
         type="text"      
     />
</div>,
]
`;
